internal/lpastore/client_test.go

```diff
@@ -0,0 +1,400 @@
+			}
+
+			return assert.Equal(t, ctx, req.Context()) &&
+				assert.Equal(t, http.MethodPut, req.Method) &&
+				assert.Equal(t, "http://base/lpas/M-0000-1111-2222", req.URL.String()) &&
+				assert.Equal(t, "application/json", req.Header.Get("Content-Type")) &&
+				assert.Equal(t, "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJvcGcucG9hcy5tYWtlcmVnaXN0ZXIiLCJzdWIiOiJ0b2RvIiwiaWF0Ijo5NDY3ODIyNDV9.6XFN3faS16wZf0garTwR4NSBxjFrAGKK3I04nK0ItMk", req.Header.Get("X-Jwt-Authorization")) &&
+				assert.JSONEq(t, expectedBody, string(body))
+		})).
+		Return(&http.Response{StatusCode: http.StatusCreated, Body: io.NopCloser(strings.NewReader(""))}, nil)
+
+	client := New("http://base", secretsClient, doer)
+	client.now = func() time.Time { return time.Date(2000, time.January, 2, 3, 4, 5, 6, time.UTC) }
+	err := client.SendLpa(ctx, donor)
+
+	assert.Nil(t, err)
+}
+
+func TestClientSendLpaWhenNewRequestError(t *testing.T) {
+	client := New("http://base", nil, nil)
+	err := client.SendLpa(nil, &actor.DonorProvidedDetails{})
+
+	assert.NotNil(t, err)
+}
+
+func TestClientSendLpaWhenSecretsClientError(t *testing.T) {
+	ctx := context.Background()
+
+	secretsClient := newMockSecretsClient(t)
+	secretsClient.
+		On("Secret", mock.Anything, mock.Anything).
+		Return("", expectedError)
+
+	client := New("http://base", secretsClient, nil)
+	err := client.SendLpa(ctx, &actor.DonorProvidedDetails{})
+
+	assert.Equal(t, expectedError, err)
+}
+
+func TestClientSendLpaWhenDoerError(t *testing.T) {
+	ctx := context.Background()
+
+	secretsClient := newMockSecretsClient(t)
+	secretsClient.
+		On("Secret", mock.Anything, mock.Anything).
+		Return("secret", nil)
+
+	doer := newMockDoer(t)
+	doer.
+		On("Do", mock.Anything).
+		Return(nil, expectedError)
+
+	client := New("http://base", secretsClient, doer)
+	err := client.SendLpa(ctx, &actor.DonorProvidedDetails{})
+
+	assert.Equal(t, expectedError, err)
+}
+
+func TestClientSendLpaWhenStatusCodeIsNotCreated(t *testing.T) {
+	ctx := context.Background()
+
+	secretsClient := newMockSecretsClient(t)
+	secretsClient.
+		On("Secret", mock.Anything, mock.Anything).
+		Return("secret", nil)
+
+	doer := newMockDoer(t)
+	doer.
+		On("Do", mock.Anything).
+		Return(&http.Response{StatusCode: http.StatusBadRequest, Body: io.NopCloser(strings.NewReader("hey"))}, nil)
+
+	client := New("http://base", secretsClient, doer)
+	err := client.SendLpa(ctx, &actor.DonorProvidedDetails{})
+
+	assert.Equal(t, responseError{name: "expected 201 response but got 400", body: "hey"}, err)
+}
+
+func TestClientServiceContract(t *testing.T) {
+	pact := &dsl.Pact{
+		Consumer:          "modernising-lpa",
+		Provider:          "data-lpa-store",
+		Host:              "localhost",
+		PactFileWriteMode: "merge",
+		LogDir:            "../../logs",
+		PactDir:           "../../pacts",
+	}
+	defer pact.Teardown()
+
+	pact.
+		AddInteraction().
+		Given("The lpa store is available").
+		UponReceiving("A request for a new case").
```

Comment:
Should we qualify the expected response by using something like:

```suggestion
		UponReceiving("A request for a new case using an existing UID").
```

Or does that not really matter? Its been a while since I used Pact so have forgotten most of the do's and don'ts.

Reply:
I will be adding more of these when I add the other data, so will expand them then